bigtable/client/readrowsparser_test.cc

```diff
@@ -0,0 +1,2054 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bigtable/client/readrowsparser.h"
+
+#include <google/protobuf/text_format.h>
+#include <gtest/gtest.h>
+
+#include <numeric>
+#include <sstream>
+#include <vector>
+
+using google::bigtable::v2::ReadRowsResponse_CellChunk;
+
+namespace bigtable {
+namespace {
+
+std::string LabelsToString(const std::vector<std::string>& labels) {
+  if (labels.empty()) return "";
+  return std::accumulate(
+      std::next(labels.begin()), labels.end(), labels[0],
+      [](std::string a, std::string b) { return a + ',' + b; });
+}
+
+// Can also be used by gtest to print Cell values
+void PrintTo(const Cell& c, ::std::ostream* os) {
+  *os << "rk: " << std::string(c.row_key()) << "\n";
+  *os << "fm: " << std::string(c.family_name()) << "\n";
+  *os << "qual: " << std::string(c.column_qualifier()) << "\n";
+  *os << "ts: " << c.timestamp() << "\n";
+  *os << "value: " << std::string(c.value()) << "\n";
+  *os << "label: " << LabelsToString(c.labels()) << "\n";
+}
+
+std::string CellToString(const Cell& cell) {
+  std::stringstream ss;
+  PrintTo(cell, &ss);
+  return ss.str();
+}
+
+};  // namespace
+};  // namespace bigtable
```

Comment:
closing braces for namespaces do not need semicolons.

Reply:
done